File: src/main/java/com/svantle/score/utils/ConfigHandler.java
package com.svantle.score.utils;

import com.svantle.score.SCore;
import org.bukkit.configuration.file.FileConfiguration;

public class ConfigHandler {

    private final SCore plugin;
    private FileConfiguration config;

    public ConfigHandler(SCore plugin) {
        this.plugin = plugin;
        loadConfig();
    }

    public FileConfiguration getConfig() {
        return config;
    }

    public void loadConfig() {
        plugin.saveDefaultConfig();
        plugin.reloadConfig();
        this.config = plugin.getConfig();
    }

}

File: src/main/kotlin/com/svantle/score/handlers/ConfigHandler.kt
package com.svantle.score.handlers

import com.svantle.score.SCore
import org.bukkit.configuration.file.FileConfiguration

class ConfigHandler(plugin: SCore) {

	val fileConfig: FileConfiguration

	init {
		plugin.saveDefaultConfig()
		plugin.reloadConfig()

		fileConfig = plugin.getConfig()
	}

	inline fun <reified T : Any> get(string: String): T {
		return fileConfig.get(string) as T
	}
}